src/pages/AnalyticsPage/Chart/Chart.tsx

```diff
@@ -0,0 +1,137 @@
+import { format } from 'date-fns'
+import { Area, AreaChart, CartesianGrid, XAxis, YAxis } from 'recharts'
+
+import {
+  type ChartConfig,
+  ChartContainer,
+  ChartLegend,
+  ChartLegendContent,
+  ChartTooltip,
+  ChartTooltipContent,
+} from 'ui/Chart'
+import Icon from 'ui/Icon'
+import LoadingLogo from 'ui/LoadingLogo'
+
+import { useCoverage } from './useCoverage'
+
+// exporting for testing
+export const formatDate = (date: string) =>
+  format(new Date(date), 'MMM d, yyyy')
+
+interface ChartProps {
+  params: {
+    startDate: Date | null
+    endDate: Date | null
+    repositories: string[]
+  }
+}
+
+function Chart({ params }: ChartProps) {
+  const {
+    data: coverage,
+    isPreviousData,
+    isLoading,
+    isError,
+  } = useCoverage({
+    params: {
+      startDate: params.startDate ? params.startDate : null,
+      endDate: params.endDate ? params.endDate : null,
+      repositories: params.repositories,
+    },
+    options: {
+      suspense: false,
+      keepPreviousData: true,
+    },
+  })
+
+  if (!isPreviousData && isLoading) {
+    return (
+      <div className="flex h-64 items-center justify-center">
+        <LoadingLogo />
+      </div>
+    )
+  }
+
+  if (isError) {
+    return (
+      <div className="flex max-h-[260px] min-h-[200px] w-full items-center justify-center">
+        <p>The coverage chart failed to load.</p>
+      </div>
+    )
+  }
+
+  if (coverage.length < 2) {
```

Comment:
I had a conversation with Joe about this at some point. We thought it made sense to render a flat line when theres exactly one data point, rather than not enough coverage message. Wdyt?

Reply:
@calvin-codecov is going to do some work on this after this PR is merged (same for the other chart)

Reply:
Ok cool sg

Reply:
@Adal3n3 and I agreed that that also makes sense. One data point is better than nothing